x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/SpatialCoordinateTypes.java

```diff
@@ -9,14 +9,16 @@
 
 import org.apache.lucene.geo.GeoEncodingUtils;
 import org.apache.lucene.geo.XYEncodingUtils;
+import org.apache.lucene.util.BytesRef;
 import org.elasticsearch.common.geo.GeoPoint;
 import org.elasticsearch.common.geo.SpatialPoint;
 import org.elasticsearch.geometry.Geometry;
 import org.elasticsearch.geometry.Point;
 import org.elasticsearch.geometry.utils.GeometryValidator;
+import org.elasticsearch.geometry.utils.WellKnownBinary;
```

Comment:
I think this enum here is kind of at the core of the implementation - we "read" the wkb using this. I'd prefer to have something that can manage to allocate less, but that can come with time.

Reply:
Yeah, this enum is collecting some cruft as we refactor the code, but I think we'll have the opportunity to cleanup q bit here once we have geo_shape completed and know the full scope of what is really needed here. The main pain is with the docs-values using such different encoding between the four main types... that is such an unnecessary hassle! But we cannot escape it because Lucene enforces that.